connectors/src/connectors/github/temporal/activities.ts

```diff
@@ -153,10 +157,152 @@ export async function githubUpsertIssueActivity(
       repoId: repoId.toString(),
       issueNumber,
       connectorId: connector.id,
+      isDiscussion: false,
     });
   }
 }
 
+export async function githubUpsertDiscussionActivity(
+  installationId: string,
+  repoName: string,
+  repoId: number,
+  login: string,
+  discussionNumber: number,
+  dataSourceConfig: DataSourceConfig,
+  loggerArgs: Record<string, string | number>
+) {
+  const localLogger = logger.child({
+    ...loggerArgs,
+    discussionNumber,
+  });
+
+  localLogger.info("Upserting GitHub discussion.");
+
+  const discussion = await getDiscussion(
+    installationId,
+    repoName,
+    login,
+    discussionNumber
+  );
+
+  let renderedDiscussion = `# ${discussion.title}||\n${discussion.bodyText}||\n`;
+
+  let nextCursor: string | null = null;
+
+  for (;;) {
+    const { cursor, comments } = await getDiscussionCommentsPage(
+      installationId,
+      repoName,
+      login,
+      discussionNumber,
+      nextCursor
+    );
+
+    for (const comment of comments) {
```

Comment:
Not parallelizing here. We can increase the number of concurrent issues/discussions that are processed when doing the full repo sync, so it's easier to assume a single job isn't doing more than 1 concurrent request.

Reply:
roger that. Worth a comment saying exactly that :+1: